src/Ombi/ClientApp/app/services/request.service.ts

```diff
@@ -2,21 +2,43 @@ import { PlatformLocation } from "@angular/common";
 import { Injectable } from "@angular/core";
 
 import { HttpClient } from "@angular/common/http";
-import { Observable } from "rxjs";
+import { Observable, ReplaySubject } from "rxjs";
 
 import { TreeNode } from "primeng/primeng";
 import { FilterType, IChildRequests, IFilter, IMovieRequestModel, IMovieRequests, IMovieUpdateModel, IRequestEngineResult, IRequestsViewModel,  ITvRequests, ITvUpdateModel, OrderType } from "../interfaces";
 import { ITvRequestViewModel } from "../interfaces";
 import { ServiceHelpers } from "./service.helpers";
 
+import { IRemainingRequests } from "../interfaces/IRemainingRequests";
+
 @Injectable()
 export class RequestService extends ServiceHelpers {
+    private requestEvents = new ReplaySubject<IRequestEngineResult>(); 
     constructor(http: HttpClient, public platformLocation: PlatformLocation) {
         super(http, "/api/v1/Request/", platformLocation);
     }
 
+    public onRequested(): Observable<IRequestEngineResult> {
+        return this.requestEvents.asObservable();
+    }
+
+    public getRemainingMovieRequests(): Observable<IRemainingRequests> {
+        return this.http.get<IRemainingRequests>(`${this.url}movie/remaining`, {headers: this.headers});
+    }
+
+    public getRemainingTvRequests(): Observable<IRemainingRequests> {
+        return this.http.get<IRemainingRequests>(`${this.url}tv/remaining`, {headers: this.headers});
+    }
+
     public requestMovie(movie: IMovieRequestModel): Observable<IRequestEngineResult> {
-        return this.http.post<IRequestEngineResult>(`${this.url}Movie/`, JSON.stringify(movie),  {headers: this.headers});
+        const observer = Observable.create(observer => {
```

Comment:
why are we doing this?

Reply:
I needed a way of passing a message to the `remainingrequests` component after a request is made, so it could update the number of requests that it's displaying. Do you think there's a cleaner way of doing this?

Reply:
Yeah personally i'd use an `EventEmitter` to emit a new event that the `remainingrequests` component can subscribe to. That's the usual way that data would get passed around in the frontend.

Reply:
Cool - I'll look into changing that.